src/Umbraco.Web/Editors/MacroController.cs

```diff
@@ -49,7 +49,9 @@ public IEnumerable<MacroParameter> GetMacroParameters(int macroId)
         /// 
         /// </param>
         /// <returns></returns>
-        public HttpResponseMessage GetMacroResultAsHtmlForEditor(string macroAlias, int pageId, [FromUri]IDictionary<string, object> macroParams)
+        [HttpPost]
+        [HttpGet]
+        public HttpResponseMessage GetMacroResultAsHtmlForEditor(string macroAlias, int pageId, [FromBody]IDictionary<string, object> macroParams)

```

Comment:
Not sure but if you use `FromBody` could you still use a GET request then? If not I'd duplicate this method and make a GET and a POST one. That would maintain backwards compatibility in case people are using the GET version.